src/main/java/edu/kit/compiler/transform/TransformContext.java

```diff
@@ -0,0 +1,175 @@
+package edu.kit.compiler.transform;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import edu.kit.compiler.data.DataType;
+import edu.kit.compiler.data.ast_nodes.ClassNode;
+import edu.kit.compiler.data.ast_nodes.MethodNode;
+import edu.kit.compiler.data.ast_nodes.MethodNode.MethodNodeParameter;
+import edu.kit.compiler.transform.TypeMapper.ClassEntry;
+import firm.Construction;
+import firm.Entity;
+import firm.Graph;
+import firm.Mode;
+import firm.Type;
+import firm.nodes.Block;
+import firm.nodes.Node;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+/*
+* Contains the necessary Context for the transformation step on a specific method.
+*
+* Precondition: Local variables of the method are counted and mapped.
+*/
+@AllArgsConstructor
+public class TransformContext {
+
+    /**
+     * Mapping of local variable names to indizes.
+     */
+    private Map<Integer, Integer> variableMapping;
+
+    /**
+     * Mapping of method parameter names to indizes.
+     */
+    private Map<Integer, Integer> paramMapping;
+
+    /**
+     * Firm types of the method parameters.
+     */
+    private Type[] paramTypes;
+
+    /**
+     * Node that projects the method parameters.
+     */
+    private Node projArgs;
+
+    @Getter
+    private TypeMapper typeMapper;
+    @Getter
+    private ClassNode classNode;
+    @Getter
+    private MethodNode methodNode;
+    @Getter
+    private boolean isStatic;
+
+    /**
+     * Return type of the method or Optional.empty() for void methods.
+     */
+    @Getter
+    private Optional<Type> returnType;
+
+    /**
+     * Current construction object.
+     */
+    @Getter
+    private Construction construction;
+
+    public TransformContext(TypeMapper typeMapper, ClassNode classNode, MethodNode methodNode,
+                            Map<Integer, Integer> variableMapping, int n_vars, boolean isStatic) {
```

Comment:
Is the `n_vars` parameter still necessary? It should always be the same as `variableMapping.size()`.

Reply:
Well, I wasn't sure whether we want to always use it this way or leave it open to the LocalVariableCounter. However, the counter doesn't return an explicit size either, so there's probably no reason to keep it